### ISSUE TITLE
Related to Research Paper

### ISSUE THREAD

**Issue by nephophile** [SY-Xuan/IICS#14]
Hi!

Q1: Could you please share your multi-branch network diagram? I get confused at many points and the network diagram may clarify many things (There is a diagram mentioned on page 3 (figure 2) but that still doesn’t clarify). Like I have this confusion that:

  a)	In Introduction it is stated that “We alleviate the domain gap by enhancing the generalization ability of trained classifiers in the first stage. Specifically, we classify each sample with C classifiers, and use their classification scores as a new feature vector..”
  So, do you mean that there is a CNN backbone that first extracts the features of every given sample sequentially and then each sample is passed on to the respective branch (in multi-branch network) and after the classification of that specific sample, the classification scores for all the samples on that camera is treated as a vector?
OR
Do you mean that after extracting the features through the backbone CNN, each sample is passed on to each branch and classification score on each branch, for each sample will be used in inter camera stage?

b)	In 3.1 it is stated that “eq. (4) can be computed on different cameras by sharing the same f..” Does it mean that f of an image will be sent to the respective branch (camera-branch) or it will be sent to all the branches? As the “sharing the same f..” is stated.

c)	In 3.3 it is also mentioned that “The classification score sm is acquired by concatenating the classification scores from C classifiers..”..
If you could please clarify the above confusions with the help of a network diagram or explain the pipeline with help of diagram (figure 2), I would be grateful.

Q2: How is sim(fm, fn) computed in section 3.1, equation 5?

Q3: In 3.2, it is stated that “l(.) computes the softmax cross entropy loss on classifier outputs and the groundtruth label m..”. Whereas, in 3.1 it was mentioned that “m denotes the cluster ID”. So, is m classifier’s output or groundtruth label? And how come there is a ground truth label, isn’t it the unsupervised setting?

Q4: What does distribution of similarities in Figure 3 show? Is it for the whole dataset or for individual cameras? Could you please elaborate?

**Comment by nephophile** [comment]
@SY-Xuan Could you please clarify the above mentioned questions? I would be grateful for your response.

**Comment by SY-Xuan** [comment]
I am sorry for the late replying.
### Q1:
(a) There are two situations in our paper. The intra-camera training and the calculation of inter-camera similarity.
    - At intra-camera training, we perform clustering within each camera. Therefore, each camera corresponds to a classifier. A CNN backbone that first extracts the features of every given sample sequentially and then each sample is passed on to the respective branch (in multi-branch network, the corresponding classifier to this sample) to calculate the loss function.
    - When calculating inter-camera similarity, we will pass the feature of each sample to all the classifiers trained at intra-camera training stage, and concatenate the classification scores to calculate the jaccard similarity.
(b) As mentioned in (a), eq. (4) can be computed on different cameras by sharing the same f..” means that f of an image will be sent to all the branches. The classification scores of all the branches are concatenated to calculate the jaccard similarity.
(c) The same as (a) and (b).
### Q2:
The first term of Eq. (5) is the cosine similarity between two features. The second term of Eq. (5) is calculated using Eq. (9).
### Q3:
This is an unsupervised setting. The m is the pseudo-label generated by clustering.
### Q4:
This is the distribution of similarity on the whole dataset. Blue and Red color indicates the distribution of similarity between
samples of the same identity from the same camera and different cameras, respectively. Yellow and Green color indicates the distribution of similarity between samples of different identities from the same camera and different cameras, respectively.

Some parts you can refer to our code. I think the code is more clear. After all, talk is cheap. lol

**Comment by nephophile** [comment]
Thank you for your response!

In **Q4**, if we take for instance the blue colored distribution, how does it indicate one bell curve for all the identities (similarity between samples of the same identities of the same cameras)?
lol. I'll definitely go through the code, as well. Thanks!

**Comment by SY-Xuan** [comment]
Yes. It is for all the identities.

**Comment by ElsaLuz** [comment]
Your responses clarified many things. Thank you!
